src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs

```diff
@@ -11,19 +11,33 @@ namespace Microsoft.AspNetCore.Mvc
 {
     public class ObjectResult : ActionResult, IStatusCodeActionResult
     {
+        private MediaTypeCollection _contentTypes;
+
         public ObjectResult(object value)
         {
             Value = value;
             Formatters = new FormatterCollection<IOutputFormatter>();
-            ContentTypes = new MediaTypeCollection();
+            _contentTypes = new MediaTypeCollection();
         }
 
         [ActionResultObjectValue]
         public object Value { get; set; }
 
         public FormatterCollection<IOutputFormatter> Formatters { get; set; }
 
-        public MediaTypeCollection ContentTypes { get; set; }
+        public MediaTypeCollection ContentTypes
+        {
+            get => _contentTypes;
+            set
+            {
+                if (_contentTypes == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
```

Comment:
Is this a breaking change? It's weird, but you never know... Might be better to just keep handling `null`?

Reply:
The `DefaultOutputFormatterSelector` doesn't allow `null` values (https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultOutputFormatterSelector.cs#L62-L65) so setting it to `null` does not work in the default code path.

Reply:
Ah, nice catch! 👌🏻